### ISSUE TITLE
Warning for simultaneous usage of both Composition API and Options API

### ISSUE THREAD

**Issue by AjiTae** [vuejs/eslint-plugin-vue#2261]
**Please describe what the rule should do:**
There really ought to be an "essential" rule that issues a warning whenever a component has both "setup" and "methods"/"watch"/"computed" sections. 

**What category should the rule belong to?**

[ ] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[X] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
export default defineComponent({
  setup() {
    return {
      x: ref(2)
    }
  },
  methods: {
    setOne() {
      this.x = 1;
    }
  }
})
```

**Additional context**
Nowadays, I see many newcomers creating frankenstein components, and it truly pains me to witness it. It would be great if such a rule could help guide them towards cleaner and more maintainable code.

**Comment by FloEdelmann** [comment]
Hmm, we already have the [`vue/component-api-style`](https://eslint.vuejs.org/rules/component-api-style.html) rule which allows enforcing one style or another.

Maybe we could add a new "consistent" option there and make it the default (as a breaking change in the next major version)?

**Comment by FloEdelmann** [comment]
But do note that using both together is actually allowed (albeit not optimal):

> You can access Composition-API-exposed values from Options API, but not the other way around.

https://vuejs.org/api/composition-api-setup.html#basic-usage

**Comment by AjiTae** [comment]
> But do note that using both together is actually allowed (albeit not optimal):
> 
> > You can access Composition-API-exposed values from Options API, but not the other way around.
> 
> https://vuejs.org/api/composition-api-setup.html#basic-usage
You can, but you certainly shouldn't.:)
